scipy/optimize/_optimize.py

```diff
@@ -3569,6 +3604,27 @@ def __call__(self, x):
         return self.f(np.asarray(x).flatten(), *self.args)
 
 
+def _make_grid_integral(grid):
+    """
+    Round a 1D grid to integers, keeping only unique integers in original range
+
+    Parameters
+    ----------
+    grid : 1D array
+
+    Returns
+    -------
+    temp : 1D array
+
+    """
+    temp = np.unique(np.round(grid))
```

Comment:
The grid start and end may not be sorted, so this code will fail for a range of (5, 0). I don't see specification like this as a big issue, how about we just sort here?

I note that
```
res = brute(rosen, [(0, 5), (5, 0)], Ns=3, finish=False)
```
works. So people may be using `brute` with `start > end`.